atividade04/src/main/java/br/edu/uepb/eda/atividade04/TabelaHash.java

```diff
@@ -0,0 +1,95 @@
+package br.edu.uepb.eda.atividade04;
+
+public class TabelaHash implements TabelaHash_IF{
+
+    private ListaEncadeada[] table;
+    private int size;
+
+    public TabelaHash(int size) {
+        this.size = size;
+        this.table = new ListaEncadeada[size];
+        for (int i = 0; i < size; i++) {
+            table[i] = new ListaEncadeada();
+        }
+    }
+
+    @Override
+    public void insert(Integer element) {
+        try {
+            table[element % size].search(element);
```

Comment:
```suggestion
            table[hash(element, size)].search(element);
```

Seria interessante centralizar a regra de hash do algoritmo, assim ficariam mais fácil dar manutenção quando precisar alterar a class